src/datachain/lib/udf.py

```diff
@@ -429,15 +458,22 @@ def _prepare_rows(udf_inputs) -> "abc.Generator[Sequence[Any], None, None]":
                         row, udf_fields, catalog, cache, download_cb
                     )
 
+        def _process_row(row):
+            with safe_closing(self.process_safe(row)) as result_objs:
+                for result_obj in result_objs:
+                    udf_output = self._flatten_row(result_obj)
+                    yield dict(zip(self.signal_names, udf_output))
```

Comment:
We need to close the iterable properly. This is because the `process_safe()` could be a generator, and might still have a file open that prevents us from removing that file.

We cannot rely on garbage collection as it might happen at a later time, so we need to close it explicitly.